### ISSUE TITLE
Upgrade to Jetty 11

### ISSUE THREAD

**Issue by DavidWhitlock** [JoyOfCodingPDX/JoyOfCoding#404]
While working on #402, I attempted to upgrade the `jetty-maven-plugin` to the latest release, `11.0.9`.  However, the `web` application wouldn't start because of:

```
Caused by: java.lang.IllegalStateException: class com.google.inject.servlet.GuiceFilter is not a jakarta.servlet.Filter
```

`GuiceFilter` indeed appears to not implement the jakarta `Filter`: https://github.com/google/guice/blob/master/extensions/servlet/src/com/google/inject/servlet/GuiceFilter.java

I'm not sure what to do here.  Maybe I can write my own jakarta `Filter` that wraps a `GuiceFilter`?

In the meantime, I'll stay on Jetty 9.

**Comment by DavidWhitlock** [comment]
Some discussion at https://github.com/google/guice/issues/1490

**Comment by DavidWhitlock** [comment]
It looks like Guice 7.0 supports the Jakarta APIs: https://github.com/google/guice/wiki/Guice700

But it looks like the resteasy-guice module was removed: https://issues.redhat.com/browse/RESTEASY-3329

**Comment by DavidWhitlock** [comment]
https://github.com/resteasy/resteasy-guice is available, which is meant to work with Guice 7.0.  Version 1.0.0.Alpha1 is available in Maven Central.  Let's give it a try.

**Comment by DavidWhitlock** [comment]
Here's what I had to do to get this working on https://github.com/JoyOfCodingPDX/JoyOfCoding/tree/issue-404/jetty-11

- Upgrade the Jetty plugin
- Replace the `javax.servlet-api` dependency with `jakarta.servlet-api`
- Replace `javax.servlet` imports with `jakarta.servlet`
- Upgrade Apache Commons File Upload to the version 2.X series.  Changed the code in `FileUploadServlet` to match the new APIs.
